ibis/bigquery/tests/test_client.py

```diff
@@ -0,0 +1,97 @@
+import uuid
+
+import numpy as np
+import pytest
+import pandas as pd
+import pandas.util.testing as tm
+import google.datalab.bigquery as bq
+
+import ibis
+import ibis.expr.types as ir
+from ibis.bigquery.tests import conftest as conftest
+
+
+def test_table(client_with_table):
+    # table must exist
+    table = client_with_table.table(conftest.TABLE)
+    assert isinstance(table, ir.TableExpr)
+
+
+def test_array_execute(table, df):
+    col_name = df.select_dtypes(['float']).columns[0]
+    expr = table[col_name]
+    result = expr.execute()[col_name]
+    expected = df[col_name]
+    tm.assert_series_equal(result, expected)
+
+
+def test_literal_execute(client):
+    expected = '1234'
+    expr = ibis.literal(expected)
+    result = client.execute(expr).iloc[0, 0]
+    assert result == expected
+
+
+def test_simple_aggregate_execute(table, df):
+    col_name = df.select_dtypes(['float']).columns[0]
+    expr = table[col_name].sum()
+    result = expr.execute()
+    expected = df[col_name].sum()
+    np.testing.assert_allclose(result, expected)
+
+
+def test_list_tables(client_with_table):
+    assert len(client_with_table.list_tables(like=conftest.TABLE)) == 1
+
+
+def test_database_layer(client_with_table):
+    bq_dataset = client_with_table._proxy.get_dataset(
+        client_with_table.dataset_id)
+    actual = client_with_table.list_tables()
+    expected = [el.name.table_id for el in bq_dataset.tables()]
+    assert sorted(actual) == sorted(expected)
```

Comment:
I would hard code the exact names you expect to be there and assert that they are in fact there, otherwise this is just copying the implementation in a test.

Reply:
I copied this from the [postgres tests](https://github.com/ibis-project/ibis/blob/v0.11.2/ibis/sql/postgres/tests/test_client.py#L72)
```
assert db.list_tables() == self.con.list_tables()
```

Reply:
I'll need to revisit that then :)

Reply:
Hm, so what I'm actually referring to here is the use of the `_proxy` API, which is internal to ibis. You should do what the postgres tests do which is construct a `database` and then confirm that the database has the same tables as the client.

Reply:
I think I'm doing the same thing [here](https://github.com/tsdlovell/ibis/blob/bigquery-backend/ibis/bigquery/client.py#L58)
```
    def get_dataset(self, dataset_id):
        return bq.Dataset(dataset_id, self.context)
```